src/app/shared/serialization/serializer.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+
+import { Base64 } from 'js-base64';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SerializerService {
+
+  constructor() {}
+
+  /**
+   * @description Return the given object serilized
+   *
+   * @remark
+   * The given object is serialized using JSON and Base64,
+   * it can be transfered usng the URL without encodeURI()
+   * and can't be human readed.
+   *
+   * @param {object} objectInsatnce
+   * @returns {string}
+   * @memberof SerializerService
+   */
+  serialize(objectInsatnce: object): string {
+
+    const serialized = Base64.encode(JSON.stringify(objectInsatnce));
+
+    return serialized;
+  }
+
+  /**
+   * @description Return an instance of the given serialized object
+   *
+   * @param {string} serializedObject
+   * @returns {object}
+   * @memberof SerializerService
+   */
+  deserialize(serializedObject: string): object {
+
+    const decodedValue: string = Base64.decode(serializedObject);
+    const deserializedObject: object = JSON.parse(decodedValue);
```

Comment:
We not use a lot the `object` type in typescript, you will have some problem with him.

Prefer to define a specific type when you can (like i describe you upper), or use the `any` type (use it less as possible)

https://stackoverflow.com/questions/18961203/any-vs-object